GVRf/Extensions/3DCursor/3DCursorLibrary/src/main/java/org/gearvrf/io/cursor3d/LaserCursor.java

```diff
@@ -55,7 +57,11 @@ void dispatchSensorEvent(SensorEvent event) {
         cursorEvent.setCursorPosition(getPositionX(), getPositionY(), getPositionZ());
         cursorEvent.setCursorRotation(getRotationW(), getRotationX(), getRotationY(),
                 getRotationZ());
-        cursorEvent.setKeyEvent(event.getCursorController().getKeyEvent());
+        GVRCursorController controller = event.getCursorController();
+        cursorEvent.setMotionEvents(controller.getMotionEvents());
```

Comment:
Though not ideal, shouldn't the event get a copy of the list? The controller's processedMotionEvent, which the event will reference to, can be updated. It is unlikely thread-safe too.